ee/tabby-db/src/users.rs

```diff
@@ -203,6 +206,16 @@ impl DbConn {
 
         Ok(())
     }
+
+    pub async fn update_user_active(&self, id: i32, active: bool) -> Result<()> {
```

Comment:
```suggestion
    pub async fn toggle_user_active(&self, id: i32) -> Result<()> {
```

Reply:
There's a few reasons I think `update` should be preferred to `toggle`. A toggle requires two database operations, one to read the current value and one to write. The frontend will already have the information about whether the user is disabled, and can just invert that and call update. This can also help keep things in sync; if two admins have an active user on their screen and both click a button to toggle whether that user is active, the user would be toggled twice, left in the same state it started in, and this would be unexpected behavior for both admins. If each clicks it and they both call update instead of toggle, the user will be left disable, and I believe this behavior is more in line with what an admin would expect.

Reply:
you can do update in a single database operation? something like `UPDATE .... SET active = !active`?

> if two admins have an active user on their screen and both click a button to toggle whether that user is active, the user would be toggled twice

Good point, we could pass in the expect current value of user for the purpose (e.g `UPDATE ... SET active = !active WHERE ...`)

Reply:
At that point, we're back to `update_user_active` except you would be supplying the *opposite* of the value you want set. I think it'd be more clear to just leave it as passing the value you want set.

Reply:
that works, but i'd really like to throw an error when you're setting same value for `active` status. That usually means there's an frontend displaying error for an critical operations like this. I'm fine to making it named as `update_user_active`, as long as it checks the current value.

Reply:
Sounds good, I'll do that in a bit